org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/coloring/ColoringParamsExtensions.xtend

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2016 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server.coloring
+
+import org.eclipse.lsp4j.ColoringParams
+import org.eclipse.xtext.resource.XtextResource
+
+/**
+ * Extension for creating empty {@link ColoringParams coloring parameter}
+ * instances or checking whether existing ones are
+ * {@link ColoringParamsExtensions#isEmpty empty} or not.
+ * 
+ * @author akos.kitta - Initial contribution and API
+ */
+abstract class ColoringParamsExtensions {
```

Comment:
I made the properties in ColoringParams and ColoringInformation nonNull and initialized the lists with an empty list. With changing the service interface to return a List of ColoringInformation, could we get rid of this extension class?